File: app/src/main/java/com/rajendra/bhajanaarti/Adapters/SongInfoAdapter.java
package com.rajendra.bhajanaarti.Adapters;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.rajendra.bhajanaarti.Pojo.SongInfo;
import com.rajendra.bhajanaarti.R;

import java.util.List;


public class  SongInfoAdapter extends ArrayAdapter<SongInfo>
{
    Context context;

    public SongInfoAdapter(Context context, int resourceId, List<SongInfo> items)
    {
        super(context, resourceId, items);
        this.context = context;
    }

    @Override
    public View getView(final int position, View view, ViewGroup parent) {

        ViewHolder holder = null;
        final SongInfo songInfo = getItem(position);

        LayoutInflater mInflater = (LayoutInflater) context.getSystemService(Activity.LAYOUT_INFLATER_SERVICE);
        if (view == null)
        {
            view = mInflater.inflate(R.layout.home_list_view, null);
            holder = new ViewHolder();
            holder.ivDeviFace = (ImageView) view.findViewById(R.id.ivDeviFace);
            holder.tvSongName = (TextView) view.findViewById(R.id.tvSongName);

            view.setTag(holder);
        }
        else
            holder = (ViewHolder) view.getTag();
            holder.ivDeviFace.setImageResource(songInfo.getImageid());
            holder.tvSongName.setText(songInfo.getSongname());

        return view;
    }

    /*private view holder class*/
    private class ViewHolder
    {
        ImageView ivDeviFace;
        TextView tvSongName;
    }
}

File: app/src/main/java/com/rajendra/bhajanaarti/Adapters/SongInfoAdapter.kt
package com.rajendra.bhajanaarti.Adapters

import android.content.Context
import android.content.Intent
import android.os.Handler
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat.startActivity
import androidx.recyclerview.widget.RecyclerView

import com.rajendra.bhajanaarti.Pojo.SongInfo
import com.rajendra.bhajanaarti.R
import com.rajendra.bhajanaarti.activities.MusicPlayerActivity

class SongInfoAdapter(private val mCtx: Context, private val items: ArrayList<SongInfo>,
                      private val progressBarInterface: ProgressBarInterface) :
        RecyclerView.Adapter<SongInfoAdapter.MyViewHolder>() {

    internal var handler = Handler()
    private var songIndex: Int? = null

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        var ivDeviFace: ImageView
        var tvSongName: TextView

        init {
            itemView.setOnClickListener(this)
            this.ivDeviFace = itemView.findViewById<View>(R.id.ivDeviFace) as ImageView
            this.tvSongName = itemView.findViewById<View>(R.id.tvSongName) as TextView
        }

        override fun onClick(itemview: View) {
            progressBarInterface.showProgressBar()
            if (MusicPlayerActivity.mp != null) {
                MusicPlayerActivity.mp!!.stop()
                MusicPlayerActivity.mp = null
            }
            songIndex = adapterPosition
            Log.d("test", "position_index $adapterPosition")
            Log.d("Test", "SongInfo_name " + items[adapterPosition].songname!!)
            handler.postDelayed(r, 900)
        }
    }

    internal var r: Runnable = Runnable {
        progressBarInterface.hideProgressBar()
        val intent = Intent(mCtx, MusicPlayerActivity::class.java)
        intent.putExtra("songindex", songIndex)
        mCtx.startActivity(intent)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.home_list_view, parent, false)

        return MyViewHolder(view)
    }


    override fun onBindViewHolder(holder: MyViewHolder, listPosition: Int) {
        val imageView = holder.ivDeviFace
        val textViewName = holder.tvSongName

        imageView.setImageResource(items[listPosition].imageid)
        textViewName.setText(items[listPosition].songname)
    }


    override fun getItemCount(): Int {
        return items.size
    }

    interface ProgressBarInterface{
        fun showProgressBar()
        fun hideProgressBar()
    }
}
